cargo-fix/src/vcs.rs

```diff
@@ -0,0 +1,66 @@
+use failure::{err_msg, Error, ResultExt};
+use std::env;
+use std::process::Command;
+
+pub enum VersionControl {
+    Git,
+    Hg,
+    Fossil,
+    Pijul,
+    None,
+}
+
+impl VersionControl {
+    pub fn new() -> Self {
+        if is_in_vcs(".git").is_ok() {
+            VersionControl::Git
+        } else if is_in_vcs(".hg").is_ok() {
+            VersionControl::Hg
+        } else if is_in_vcs(".pijul").is_ok() {
+            VersionControl::Pijul
+        } else if is_in_vcs(".fossil").is_ok() {
```

Comment:
`.fossil` file can actually be in a completely unrelated directory, so I doubt this would work often, but it's the same simple idea used by `cargo-new`